design/set.behavior.actor.md

```diff
@@ -0,0 +1,33 @@
+# Set Behavior API
+
+An API is needed on the actor for the developer to be able to set a bahavior for the actor to use to 
+allow fo user interaction with that actor.  The API should ensure that the behavior is tied to the one
+actor it is being set on and should be flexible enough to allow setting params on the the behavior at the
+time of creation or initialization.
+
+## API
+
+### In actor.ts
+``` ts
+/**
+ * Sets the behavior on this actor.
+ * @param behavior The type of behavior to set. Pass null to clear the behavior.
+ * @param initOptions The init options object to pass to the behavior to initialize it.
+ */
+public setBehavior<BehaviorT extends Behavior, InitOptionsT>(
```

Comment:
This was what I had pinged you about earlier to chat about.  This will likely be refined further as I develop this feature.  We should chat offline and I will update the design as this evolves from the tech implementation side of this.  This main point here is that I want the developer to be able to pass in optional config params to the set behavior call.  Still working out how best to accomplish that.